src/Schema/Builder.php

```diff
@@ -107,6 +109,48 @@ public function dropAllTables()
         }
     }
 
+    public function getTables()
+    {
+        $db = $this->connection->getMongoDB();
+        $collections = [];
+
+        foreach ($db->listCollections() as $collectionInfos) {
+            $stats = $db->selectCollection($collectionInfos->getName())->aggregate([
+                ['$collStats' => ['storageStats' => ['scale' => 1]]],
+                ['$project' => ['storageStats.totalSize' => 1]],
+            ])->toArray();
+
+            $collections[] = [
+                'name' => $collectionInfos->getName(),
```

Comment:
Consider using `$db->listCollectionNames()` if you're only going to use the name. It passes the `nameOnly` option to the `listCollections` command, which is faster. Quoting from the [`listCollections` command docs](https://www.mongodb.com/docs/manual/reference/command/listCollections/#command-fields):
> Returning just the name and type (`view`, `collection`, or `timeseries`) does not take collection-level locks whereas returning full collection information locks each collection in the database.

```suggestion
        foreach ($db->listCollectionNames() as $collectionName) {
            $stats = $db->selectCollection($collectionName)->aggregate([
                ['$collStats' => ['storageStats' => ['scale' => 1]]],
                ['$project' => ['storageStats.totalSize' => 1]],
            ])->toArray();

            $collections[] = [
                'name' => $collectionName,
```